docs/syntax/syntax.md

```diff
@@ -255,7 +255,8 @@ By default, MyST will resolve link destinations according to the following rules
 3. Destinations which point to a local file path are treated as links to that file.
    - The path must be relative and in [POSIX format](https://en.wikipedia.org/wiki/Path_(computing)#POSIX_and_Unix_paths) (i.e. `/` separators).
    - If the path is to another source file in the project (e.g. a `.md` or `.rst` file),
-    then the link will be to the initial heading in that file.
+    then the link will be to the initial heading in that file or,
+    if the path is appended by a `#target`, to the heading "slug" in that file.
```

Comment:
What happens if `#target` is also defined as a section label as well? E.g. if there's content like:

```
(target)=
## Some section

Some text

## Target

Some other text, and a ref to [](#target).
```

And does the behavior change at all if instead the reference was `[](myfile.md#target)`?

Reply:
ah ok - I thought this was implementing that part of the MEP, but if this is just a step forward in _anticipation_ of the MEP, I think we should just merge it as-is

Reply:
> I thought this was implementing that part of the MEP,

yeh, no because obviously that could start breaking peoples projects, so that needs to be done with great care 😅 (and ideally provide them a way to migrate) 

With https://github.com/executablebooks/myst-enhancement-proposals/pull/10, we would like `#target` to search through essentially all targets on that page (and possibly all pages); headings, figures, tables, equations, ...

1. You need a systematic way to do this
2. and you need to allow for a way to filter on ambiguous target names, i.e. by specifying the domain and/or object type (similar to the new `inv:key:domain:type#name` links)

In #613 I have kind of achieved this, but it does involve some "patching" of sphinx's code, for example they do some weird stuff with creating latex labels, which I guess does introduce a bit more potential maintenance burden.